src/Features/Core/Portable/InvertIf/AbstractInvertIfCodeRefactoringProvider.cs

```diff
@@ -472,10 +475,26 @@ private ImmutableArray<StatementRange> GetSubsequentStatementRanges(TIfStatement
                         var index = statements.IndexOf(ifNode);
 
                         var statementsBeforeIf = statements.Take(index);
-                        var statementsAfterIf = statements.Skip(index + 1);
+                        var statementsAfterIf = statements.Skip(index + 1).ToImmutableArray();
 
                         var ifBody = GetIfBody(ifNode);
 
+                        var newTrailing = UnwrapBlock(ifBody).ToArray();
+
+                        // Get leading and trailing space of the expressions to preserve for the user
+                        // ex:
+                        // if (true)
+                        // {
+                        //    return true;
+                        // }
+                        //              // <<< preserve this line
+                        // // preserve this comment
+                        // return false;
+                        var leadingTrivia = GetLeadingSpace(statementsAfterIf[0].GetLeadingTrivia()).Concat(GetTriviaAfterSpace(newTrailing[0].GetLeadingTrivia()));
+                        var trailingTrivia = GetTriviaUntilSpace(newTrailing[^1].GetTrailingTrivia()).Concat(GetTrailingSpace(statementsAfterIf[^1].GetTrailingTrivia()));
+                        newTrailing[0] = newTrailing[0].WithLeadingTrivia(leadingTrivia);
+                        newTrailing[^1] = newTrailing[^1].WithTrailingTrivia(trailingTrivia);
```

Comment:
unclear how you know that newTrailing has at least oen element in it.

Reply:
It was previously the body of the if statement. Let me see if there are tests to verify my assumption though. I don't think we'd make it here if it was empty

Reply:
Can you add a test where there is no trivia anywhere in the if and the statements?

Reply:
(like no spaces/newlines *at all*).

Reply:
oh you mean like...

```csharp
if(true){return true;}return false;
```

Reply:
Yup, exactly.  :)

Reply:
or even better:

```
{if(true){return(true);}return(false);}
```

No spaces/newlines anywhere.

Reply:
```csharp
return(true);
```

oh no...